client/post-editor/editor-more-options/copy-post.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'state/ui/selectors';
+import AccordionSection from 'components/accordion/section';
+import Button from 'components/button';
+import Dialog from 'components/dialog';
+import FormSectionHeading from 'components/forms/form-section-heading';
+import Gridicon from 'components/gridicon';
+import InfoPopover from 'components/info-popover';
+import PostSelector from 'my-sites/post-selector';
+
+class EditorMoreOptionsCopyPost extends Component {
+
+	static propTypes = {
+		site: PropTypes.object.isRequired,
+		translate: PropTypes.func,
+	};
+
+	constructor() {
+		super( ...arguments );
+		this.state = this.resetState();
+	}
+
+	resetState() {
+		return {
+			selectedPostId: null,
+			showDialog: false,
+		};
+	}
+
+	openDialog = event => {
+		event.preventDefault();
+
+		this.setState( {
+			showDialog: true,
+		} );
+	}
+
+	closeDialog = () => {
```

Comment:
Oh, this is just something I do on autopilot.
I've never quite liked binding in the constructor, and I feel this way is more readable.
As it's something I've started using quite some time ago, I had to check [how Babel transpiles it](https://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-0&code=class%20Foo%20%7B%0A%0A%20%20onClick%20%3D%20event%20%3D%3E%20%7B%0A%20%20%20%20console.log(%20'clicked'%20)%3B%0A%20%20%7D%0A%20%0A%7D&experimental=true&loose=false&spec=false), and it basically moves the entire arrow method into the transpiled constructor.
I believe there are no downsides about it, as long as we don't care about transpiled code readability.

A note, though: you may have noticed that the REPL is set to `stage-0`, and if you try replacing it to `stage-2` fails miserably. That's because those methods becomes actual class properties and require the `transform-class-properties` plugin - which Calypso uses (I guess for the static properties?).

Let me know it this is an unclear, unreadable or just plain wrong approach.